src/makeCalculator.js

```diff
@@ -4,7 +4,43 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculate = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (calculate.hasOwnProperty(callback) === false) {
+        throw new Error('Invalid operation');
```

Comment:
The error message 'Invalid operation' is a bit vague. It would be more helpful to specify that the provided callback is not a function, for example: 'The provided callback must be a function.'